pallets/aleph/src/lib.rs

```diff
@@ -123,6 +129,18 @@ pub mod pallet {
     pub(super) type FinalityScheduledVersionChange<T: Config> =
         StorageValue<_, VersionChange, OptionQuery>;
 
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_finalize(block_number: T::BlockNumber) {
+            if block_number % DEFAULT_SESSION_PERIOD.into() == (DEFAULT_SESSION_PERIOD - 1).into() {
```

Comment:
Question: are sessions guaranteed to last `DEFAULT_SESSION_PERIOD` or are there circumstances when they might last longer or shorter?

Reply:
Even emergency finalizer can't change that. The session length is always constant, and this constant is `DEFAULT_SESSION_PERIOD` (the value is hardcoded into the runtime, the only way to change it is the `short_session` feature flag, or changing it explicitly in the code).

Reply:
Right, finalizer does not change that, what I wrote does not really make sense

Reply:
The check above is fine but a little nicer would be to use https://paritytech.github.io/substrate/master/pallet_session/pallet/trait.Config.html#associatedtype.NextSessionRotation which in our case is `PeriodicSession` instantiated here:     type ShouldEndSession = pallet_session::PeriodicSessions<SessionPeriod, Offset>;